### PR TITLE
zkapp transaction limits in CI

### PR DESCRIPTION
**Pull request by deepthiskumar**
Show 
* possible account update combinations
* event limits
* Action limits

Add Proof and verification key to the heap_usage app

Example output https://buildkite.com/o-1-labs-2/mina-o-1-labs/builds/28045#0187a5c4-3de0-4960-92e8-2d99a344a9f4

```


Generated zkapp transactions with 17 updates and 16 proof updates in 43.575141 secs
--
  | Generated updates permutation 0: SPPPPPPPPPPPPPPPP
  | Updating authorizations...
  | Data of type Account.t (w/ zkapp)                 uses    211 heap words =    13504 bytes
  | Data of type Zkapp_command.t                      uses  52795 heap words =  3378880 bytes
  | Data of type Pickles.Side_loaded.Proof.t          uses   3502 heap words =   224128 bytes
  | Data of type Mina_base.Side_loaded_verification_key.t uses    890 heap words =    56960 bytes
  | Data of type Dummy Pickles.Side_loaded.Proof.t    uses   2707 heap words =   173248 bytes
  | Data of type Dummy Mina_base.Side_loaded_verification_key.t uses     99 heap words =     6336 bytes
  | Data of type Ledger.Db.path.t                     uses    422 heap words =    27008 bytes
  | Data of type Protocol_state.t                     uses    577 heap words =    36928 bytes
  | Data of type Pending_coinbase.t                   uses    595 heap words =    38080 bytes
  | Data of type Staged_ledger_diff.t (payments)      uses    730 heap words =    46720 bytes
  | Data of type Parallel_scan.Base.t (coinbase)      uses   1008 heap words =    64512 bytes
  | Data of type Parallel_scan.Base.t (payment)       uses   1158 heap words =    74112 bytes
  | Data of type Parallel_scan.Base.t (zkApp)         uses  54816 heap words =  3508224 bytes
  | Data of type Parallel_scan.Merge.t                uses   8785 heap words =   562240 bytes
  | Data of type Transaction_snark.Statement.t        uses    260 heap words =    16640 bytes
  | Building zkapp limits app
  | version: 4.14.0
  | word_size: 64
  | check_opam_switch opam.export
  | Starting Build
  | Build complete
  | ulimit -s 65532 && (ulimit -n 10240 \|\| true) && dune build src/app/zkapp_limits/zkapp_limits.exe --profile=devnet
  | Running heap usage app
  | max field elements for events per transaction: 100
  | max field elements for actions per transaction: 100
  | All possible zkApp account update combinationsProofs updates=0  Signed/None updates=0  Pairs of Signed/None updates=0: Total account updates: 0 Cost: 0.000000
  | Proofs updates=0  Signed/None updates=0  Pairs of Signed/None updates=1: Total account updates: 2 Cost: 10.080000
  | Proofs updates=0  Signed/None updates=0  Pairs of Signed/None updates=2: Total account updates: 4 Cost: 20.160000
  | Proofs updates=0  Signed/None updates=0  Pairs of Signed/None updates=3: Total account updates: 6 Cost: 30.240000
  | Proofs updates=0  Signed/None updates=0  Pairs of Signed/None updates=4: Total account updates: 8 Cost: 40.320000
  | Proofs updates=0  Signed/None updates=0  Pairs of Signed/None updates=5: Total account updates: 10 Cost: 50.400000
  | Proofs updates=0  Signed/None updates=0  Pairs of Signed/None updates=6: Total account updates: 12 Cost: 60.480000
  | Proofs updates=0  Signed/None updates=1  Pairs of Signed/None updates=0: Total account updates: 1 Cost: 9.140000
  | Proofs updates=0  Signed/None updates=1  Pairs of Signed/None updates=1: Total account updates: 3 Cost: 19.220000
  | Proofs updates=0  Signed/None updates=1  Pairs of Signed/None updates=2: Total account updates: 5 Cost: 29.300000
  | Proofs updates=0  Signed/None updates=1  Pairs of Signed/None updates=3: Total account updates: 7 Cost: 39.380000
  | Proofs updates=0  Signed/None updates=1  Pairs of Signed/None updates=4: Total account updates: 9 Cost: 49.460000
  | Proofs updates=0  Signed/None updates=1  Pairs of Signed/None updates=5: Total account updates: 11 Cost: 59.540000
  | Proofs updates=0  Signed/None updates=2  Pairs of Signed/None updates=0: Total account updates: 2 Cost: 18.280000
  | Proofs updates=0  Signed/None updates=2  Pairs of Signed/None updates=1: Total account updates: 4 Cost: 28.360000
  | Proofs updates=0  Signed/None updates=2  Pairs of Signed/None updates=2: Total account updates: 6 Cost: 38.440000
  | Proofs updates=0  Signed/None updates=2  Pairs of Signed/None updates=3: Total account updates: 8 Cost: 48.520000
  | Proofs updates=0  Signed/None updates=2  Pairs of Signed/None updates=4: Total account updates: 10 Cost: 58.600000
  | Proofs updates=0  Signed/None updates=2  Pairs of Signed/None updates=5: Total account updates: 12 Cost: 68.680000
  | Proofs updates=0  Signed/None updates=3  Pairs of Signed/None updates=0: Total account updates: 3 Cost: 27.420000
  | Proofs updates=0  Signed/None updates=3  Pairs of Signed/None updates=1: Total account updates: 5 Cost: 37.500000
  | Proofs updates=0  Signed/None updates=3  Pairs of Signed/None updates=2: Total account updates: 7 Cost: 47.580000
  | Proofs updates=0  Signed/None updates=3  Pairs of Signed/None updates=3: Total account updates: 9 Cost: 57.660000
  | Proofs updates=0  Signed/None updates=3  Pairs of Signed/None updates=4: Total account updates: 11 Cost: 67.740000
  | Proofs updates=0  Signed/None updates=4  Pairs of Signed/None updates=0: Total account updates: 4 Cost: 36.560000
  | Proofs updates=0  Signed/None updates=4  Pairs of Signed/None updates=1: Total account updates: 6 Cost: 46.640000
  | Proofs updates=0  Signed/None updates=4  Pairs of Signed/None updates=2: Total account updates: 8 Cost: 56.720000
  | Proofs updates=0  Signed/None updates=4  Pairs of Signed/None updates=3: Total account updates: 10 Cost: 66.800000
  | Proofs updates=0  Signed/None updates=5  Pairs of Signed/None updates=0: Total account updates: 5 Cost: 45.700000
  | Proofs updates=0  Signed/None updates=5  Pairs of Signed/None updates=1: Total account updates: 7 Cost: 55.780000
  | Proofs updates=0  Signed/None updates=5  Pairs of Signed/None updates=2: Total account updates: 9 Cost: 65.860000
  | Proofs updates=0  Signed/None updates=6  Pairs of Signed/None updates=0: Total account updates: 6 Cost: 54.840000
  | Proofs updates=0  Signed/None updates=6  Pairs of Signed/None updates=1: Total account updates: 8 Cost: 64.920000
  | Proofs updates=0  Signed/None updates=7  Pairs of Signed/None updates=0: Total account updates: 7 Cost: 63.980000
  | Proofs updates=1  Signed/None updates=0  Pairs of Signed/None updates=0: Total account updates: 1 Cost: 10.260000
  | Proofs updates=1  Signed/None updates=0  Pairs of Signed/None updates=1: Total account updates: 3 Cost: 20.340000
  | Proofs updates=1  Signed/None updates=0  Pairs of Signed/None updates=2: Total account updates: 5 Cost: 30.420000
  | Proofs updates=1  Signed/None updates=0  Pairs of Signed/None updates=3: Total account updates: 7 Cost: 40.500000
  | Proofs updates=1  Signed/None updates=0  Pairs of Signed/None updates=4: Total account updates: 9 Cost: 50.580000
  | Proofs updates=1  Signed/None updates=0  Pairs of Signed/None updates=5: Total account updates: 11 Cost: 60.660000
  | Proofs updates=1  Signed/None updates=1  Pairs of Signed/None updates=0: Total account updates: 2 Cost: 19.400000
  | Proofs updates=1  Signed/None updates=1  Pairs of Signed/None updates=1: Total account updates: 4 Cost: 29.480000
  | Proofs updates=1  Signed/None updates=1  Pairs of Signed/None updates=2: Total account updates: 6 Cost: 39.560000
  | Proofs updates=1  Signed/None updates=1  Pairs of Signed/None updates=3: Total account updates: 8 Cost: 49.640000
  | Proofs updates=1  Signed/None updates=1  Pairs of Signed/None updates=4: Total account updates: 10 Cost: 59.720000
  | Proofs updates=1  Signed/None updates=2  Pairs of Signed/None updates=0: Total account updates: 3 Cost: 28.540000
  | Proofs updates=1  Signed/None updates=2  Pairs of Signed/None updates=1: Total account updates: 5 Cost: 38.620000
  | Proofs updates=1  Signed/None updates=2  Pairs of Signed/None updates=2: Total account updates: 7 Cost: 48.700000
  | Proofs updates=1  Signed/None updates=2  Pairs of Signed/None updates=3: Total account updates: 9 Cost: 58.780000
  | Proofs updates=1  Signed/None updates=2  Pairs of Signed/None updates=4: Total account updates: 11 Cost: 68.860000
  | Proofs updates=1  Signed/None updates=3  Pairs of Signed/None updates=0: Total account updates: 4 Cost: 37.680000
  | Proofs updates=1  Signed/None updates=3  Pairs of Signed/None updates=1: Total account updates: 6 Cost: 47.760000
  | Proofs updates=1  Signed/None updates=3  Pairs of Signed/None updates=2: Total account updates: 8 Cost: 57.840000
  | Proofs updates=1  Signed/None updates=3  Pairs of Signed/None updates=3: Total account updates: 10 Cost: 67.920000
  | Proofs updates=1  Signed/None updates=4  Pairs of Signed/None updates=0: Total account updates: 5 Cost: 46.820000
  | Proofs updates=1  Signed/None updates=4  Pairs of Signed/None updates=1: Total account updates: 7 Cost: 56.900000
  | Proofs updates=1  Signed/None updates=4  Pairs of Signed/None updates=2: Total account updates: 9 Cost: 66.980000
  | Proofs updates=1  Signed/None updates=5  Pairs of Signed/None updates=0: Total account updates: 6 Cost: 55.960000
  | Proofs updates=1  Signed/None updates=5  Pairs of Signed/None updates=1: Total account updates: 8 Cost: 66.040000
  | Proofs updates=1  Signed/None updates=6  Pairs of Signed/None updates=0: Total account updates: 7 Cost: 65.100000
  | Proofs updates=2  Signed/None updates=0  Pairs of Signed/None updates=0: Total account updates: 2 Cost: 20.520000
  | Proofs updates=2  Signed/None updates=0  Pairs of Signed/None updates=1: Total account updates: 4 Cost: 30.600000
  | Proofs updates=2  Signed/None updates=0  Pairs of Signed/None updates=2: Total account updates: 6 Cost: 40.680000
  | Proofs updates=2  Signed/None updates=0  Pairs of Signed/None updates=3: Total account updates: 8 Cost: 50.760000
  | Proofs updates=2  Signed/None updates=0  Pairs of Signed/None updates=4: Total account updates: 10 Cost: 60.840000
  | Proofs updates=2  Signed/None updates=1  Pairs of Signed/None updates=0: Total account updates: 3 Cost: 29.660000
  | Proofs updates=2  Signed/None updates=1  Pairs of Signed/None updates=1: Total account updates: 5 Cost: 39.740000
  | Proofs updates=2  Signed/None updates=1  Pairs of Signed/None updates=2: Total account updates: 7 Cost: 49.820000
  | Proofs updates=2  Signed/None updates=1  Pairs of Signed/None updates=3: Total account updates: 9 Cost: 59.900000
  | Proofs updates=2  Signed/None updates=2  Pairs of Signed/None updates=0: Total account updates: 4 Cost: 38.800000
  | Proofs updates=2  Signed/None updates=2  Pairs of Signed/None updates=1: Total account updates: 6 Cost: 48.880000
  | Proofs updates=2  Signed/None updates=2  Pairs of Signed/None updates=2: Total account updates: 8 Cost: 58.960000
  | Proofs updates=2  Signed/None updates=2  Pairs of Signed/None updates=3: Total account updates: 10 Cost: 69.040000
  | Proofs updates=2  Signed/None updates=3  Pairs of Signed/None updates=0: Total account updates: 5 Cost: 47.940000
  | Proofs updates=2  Signed/None updates=3  Pairs of Signed/None updates=1: Total account updates: 7 Cost: 58.020000
  | Proofs updates=2  Signed/None updates=3  Pairs of Signed/None updates=2: Total account updates: 9 Cost: 68.100000
  | Proofs updates=2  Signed/None updates=4  Pairs of Signed/None updates=0: Total account updates: 6 Cost: 57.080000
  | Proofs updates=2  Signed/None updates=4  Pairs of Signed/None updates=1: Total account updates: 8 Cost: 67.160000
  | Proofs updates=2  Signed/None updates=5  Pairs of Signed/None updates=0: Total account updates: 7 Cost: 66.220000
  | Proofs updates=3  Signed/None updates=0  Pairs of Signed/None updates=0: Total account updates: 3 Cost: 30.780000
  | Proofs updates=3  Signed/None updates=0  Pairs of Signed/None updates=1: Total account updates: 5 Cost: 40.860000
  | Proofs updates=3  Signed/None updates=0  Pairs of Signed/None updates=2: Total account updates: 7 Cost: 50.940000
  | Proofs updates=3  Signed/None updates=0  Pairs of Signed/None updates=3: Total account updates: 9 Cost: 61.020000
  | Proofs updates=3  Signed/None updates=1  Pairs of Signed/None updates=0: Total account updates: 4 Cost: 39.920000
  | Proofs updates=3  Signed/None updates=1  Pairs of Signed/None updates=1: Total account updates: 6 Cost: 50.000000
  | Proofs updates=3  Signed/None updates=1  Pairs of Signed/None updates=2: Total account updates: 8 Cost: 60.080000
  | Proofs updates=3  Signed/None updates=2  Pairs of Signed/None updates=0: Total account updates: 5 Cost: 49.060000
  | Proofs updates=3  Signed/None updates=2  Pairs of Signed/None updates=1: Total account updates: 7 Cost: 59.140000
  | Proofs updates=3  Signed/None updates=2  Pairs of Signed/None updates=2: Total account updates: 9 Cost: 69.220000
  | Proofs updates=3  Signed/None updates=3  Pairs of Signed/None updates=0: Total account updates: 6 Cost: 58.200000
  | Proofs updates=3  Signed/None updates=3  Pairs of Signed/None updates=1: Total account updates: 8 Cost: 68.280000
  | Proofs updates=3  Signed/None updates=4  Pairs of Signed/None updates=0: Total account updates: 7 Cost: 67.340000
  | Proofs updates=4  Signed/None updates=0  Pairs of Signed/None updates=0: Total account updates: 4 Cost: 41.040000
  | Proofs updates=4  Signed/None updates=0  Pairs of Signed/None updates=1: Total account updates: 6 Cost: 51.120000
  | Proofs updates=4  Signed/None updates=0  Pairs of Signed/None updates=2: Total account updates: 8 Cost: 61.200000
  | Proofs updates=4  Signed/None updates=1  Pairs of Signed/None updates=0: Total account updates: 5 Cost: 50.180000
  | Proofs updates=4  Signed/None updates=1  Pairs of Signed/None updates=1: Total account updates: 7 Cost: 60.260000
  | Proofs updates=4  Signed/None updates=2  Pairs of Signed/None updates=0: Total account updates: 6 Cost: 59.320000
  | Proofs updates=4  Signed/None updates=2  Pairs of Signed/None updates=1: Total account updates: 8 Cost: 69.400000
  | Proofs updates=4  Signed/None updates=3  Pairs of Signed/None updates=0: Total account updates: 7 Cost: 68.460000
  | Proofs updates=5  Signed/None updates=0  Pairs of Signed/None updates=0: Total account updates: 5 Cost: 51.300000
  | Proofs updates=5  Signed/None updates=0  Pairs of Signed/None updates=1: Total account updates: 7 Cost: 61.380000
  | Proofs updates=5  Signed/None updates=1  Pairs of Signed/None updates=0: Total account updates: 6 Cost: 60.440000
  | Proofs updates=6  Signed/None updates=0  Pairs of Signed/None updates=0: Total account updates: 6 Cost: 61.560000

```

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Review by lk86 — APPROVED** [review]

File: src/app/zkapp_limits/zkapp_limits.ml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+let genesis_constants = Genesis_constants.compiled
+
+let main () =
+  let cost_limit = genesis_constants.zkapp_transaction_cost_limit in
+  let max_event_elements = genesis_constants.max_event_elements in
+  let max_action_elements = genesis_constants.max_action_elements in
+  let values = List.init 20 ~f:Fn.id in
+  printf
+    !"max field elements for events per transaction: %d\n"
```
Comment (psteckler) [inline src/app/zkapp_limits/zkapp_limits.ml:12]: don't need the `!`, unless you have the `%{sexp: t}` thing in the format string

File: src/app/heap_usage/values.ml [hunk 2]
```diff
@@ -41,6 +41,26 @@ let zkapp_command =
...
+    ~f:(fun _acc a ->
+      match a.Mina_base.Account_update.authorization with
+      | Proof proof ->
+          Stop (Some proof)
+      | _ ->
+          Continue None )
+    ~finish:Fn.id
+  |> Option.value_exn
+
+let verification_key =
```
Comment (psteckler) [inline src/app/heap_usage/values.ml:57]: could you just use the dummy vk, won't it be the same size?

**Comment by psteckler** [comment]
Doesn't the Dhall file need to refer to the renamed script in `buildkite/`?

File: src/app/heap_usage/values.ml (hunk 2 above)
Comment (deepthiskumar) [reply src/app/heap_usage/values.ml:63]: They were showing different sizes- Added dummy proof and vk as well to the app

File: src/app/heap_usage/values.ml [hunk 3]
```diff
@@ -41,6 +41,26 @@ let zkapp_command =
   in
   List.hd_exn zkapp_commands
 
+let zkapp_proof =
```
[the remark this answers is not in the document]
Comment (deepthiskumar) [reply src/app/heap_usage/values.ml:44]: Same as above

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Review by psteckler — APPROVED** [review]

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Pull request merged**